File: app/src/main/java/org/michaelbel/moviemade/data/di/component/ActivityComponent.java
package org.michaelbel.moviemade.data.di.component;

import org.michaelbel.moviemade.data.di.module.ActivityModule;
import org.michaelbel.moviemade.data.di.scope.ActivityScoped;
import org.michaelbel.moviemade.ui.modules.main.MainActivity;
import org.michaelbel.moviemade.ui.modules.movie.MovieActivity;
import org.michaelbel.moviemade.ui.modules.reviews.activity.ReviewActivity;

import dagger.Subcomponent;

@ActivityScoped
@Subcomponent(modules = ActivityModule.class)
public interface ActivityComponent {
    void inject(MainActivity target);
    void inject(MovieActivity target);
    void inject(ReviewActivity target);
}

File: app/src/main/java/org/michaelbel/moviemade/presentation/di/component/ActivityComponent.kt
package org.michaelbel.moviemade.presentation.di.component

import dagger.Subcomponent
import org.michaelbel.moviemade.presentation.di.module.ActivityModule
import org.michaelbel.moviemade.presentation.di.scope.ActivityScope
import org.michaelbel.moviemade.presentation.features.main.MainActivity
import org.michaelbel.moviemade.presentation.features.movie.MovieActivity

@ActivityScope
@Subcomponent(modules = [ActivityModule::class])
interface ActivityComponent {
    fun inject(target: MainActivity)
    fun inject(target: MovieActivity)
}